### PR TITLE
Add Angular "Search Results" page

### PR DESCRIPTION
**Pull request by TheGuyWhoCodes**
## Majors 
 - Add Search Results page
    - Displays results for both books and movies with pagination. 
    - Supported with tabs between books and movies
 - Adds entity helper
    - Used to quick display information about a book / movie
    - Should be refactored in the future

## Minors
 - Basic tests built for the three components

## Questions
 - @henryksloan where should the "powered by google" label go?

## Screenshots
image
image

### CODE REVIEW

File: mediaphile/src/app/pages/search/entity/entity.component.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  type: string
+
+  constructor() { }
+
+  ngOnInit(): void {
+
+  }
+
+  public getEntityImageUrl() : string {
+    if(this.type == "book") {
```
Comment (henryksloan) [inline mediaphile/src/app/pages/search/entity/entity.component.ts:23]: I think these `==`s should be `===`s

File: mediaphile/src/app/pages/search/results/results.component.html [hunk 2]
```diff
@@ -0,0 +1,9 @@
+  <h2>Exploring {{query}}: </h2>
+  <div *ngIf = "hasReceivedResults()">
```
Comment (henryksloan) [inline mediaphile/src/app/pages/search/results/results.component.html:2]: No spaces around `=` in attributes list.

**Comment by kirantumkur** [comment]
@XLuna24 @henryksloan  Please review this and provide comments or approve this!

File: mediaphile/src/app/pages/search/entity/entity.component.ts (hunk 1 above)
Comment (TheGuyWhoCodes) [reply mediaphile/src/app/pages/search/entity/entity.component.ts:23]: ah yes thanks for this!!! i always forget about the triple equals

File: mediaphile/src/app/pages/search/results/results.component.html (hunk 2 above)
Comment (TheGuyWhoCodes) [reply mediaphile/src/app/pages/search/results/results.component.html]: fixed per fc4838b

File: mediaphile/src/app/pages/search/entity/entity.component.ts (hunk 1 above)
Comment (TheGuyWhoCodes) [reply mediaphile/src/app/pages/search/entity/entity.component.ts]: fixed per fc4838b

**Pull request merged**